### PR TITLE
LPS-161475 KB Folder Actions

### PR DESCRIPTION
**Pull request by ambrinchaudhary**
Screenshot 2022-08-25 at 14 36 09

Created a separate story to show the icons in the navigation pane: https://issues.liferay.com/browse/LPS-161489

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Review by boton — CHANGES_REQUESTED** [review]
Ambrin and I discovered some actions aren't working and don't have icons.
The data for actions needs a series of transformations before passing to the dropdown.

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/normalizeDropdownItems.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+export default function normalizeDropdownItems(items) {
+	return items.map(({icon, ...props}) => ({...props, symbolLeft: icon}));
```
Comment (boton) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/normalizeDropdownItems.js:16]: Can we move the propsTransformer here? 
I think is part of normalization and we don't need a normalized DropdownItems without it.

Something like:

```js
return KBDropdownPropsTransformer(
    items.map(({icon, ...props}) => ({...props, symbolLeft: icon}))
);
```

Maybe isn't a bad idea to add a default value `[]` to items.

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/normalizeDropdownItems.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+export default function normalizeDropdownItems(items) {
```
Comment (boton) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/normalizeDropdownItems.js:15]: Maybe isn't a bad idea to add a default value `[]` to items.

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/NavigationPanel.js [hunk 3]
```diff
@@ -58,8 +68,34 @@ export default function NavigationPanel({items, selectedItemId}) {
...
+									});
+								}
+
 								return (
 									<ClayTreeView.Item
+										actions={
+											item?.actions?.length && (
+												<ClayDropDownWithItems
+													items={
+														transformedActions.items
```
Comment (boton) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/NavigationPanel.js:87]: Can we simply inline this? 
If we inline we avoid checking two times for `item?.actions?.length` and creating `transformedActions`.

Something like this (after moving propsTransformer inside of  normalizeDropdownItems)

```jsx
items={normalizeDropdownItems(item.actions)}
```

**Review by boton — CHANGES_REQUESTED** [review]
I left a few comments inline

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/normalizeDropdownItems.js (hunk 1 above)
Comment (ambrinchaudhary) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/normalizeDropdownItems.js:16]: Yeah makes sense, I had the same feeling but finally left them separated, but it is good to know you think the same. 👍

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Comment by ambrinchaudhary** [comment]
ci:test:relevant

**Review by boton — APPROVED** [review]
Tested and working nice! 🎉 

image

And I love the [normalizeDropdownItems](https://github.com/liferay-lima/liferay-portal/blob/92bbb84be82e808c2ff064eb6b10387c5f378b4c/modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/normalizeDropdownItems.js) ❤️

**Comment by boton** [comment]
ci:forward

**Comment by liuxingye** [comment]
ci:forward

**Comment by liuxingye** [comment]
ci:forward

**Comment by ambrinchaudhary** [comment]
ci:reevaluate:1267401_4790

**Comment by ambrinchaudhary** [comment]
Manually forwarded to https://github.com/brianchandotcom/liferay-portal/pull/122495

**Pull request closed without merging**